test/models/user_test.rb

```diff
@@ -1,7 +1,19 @@
-require "test_helper"
+# test/models/user_test.rb
+require 'test_helper'
 
 class UserTest < ActiveSupport::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
+  test "should not save user without first_name" do
+    user = User.new(last_name: "Doe", email: "john@example.com", password: "password")
+    assert_not user.save, "Saved the user without a first_name"
+  end
+
+  test "should not save user without last_name" do
+    user = User.new(first_name: "John", email: "john@example.com", password: "password")
+    assert_not user.save, "Saved the user without a last_name"
+  end
+
+  test "should save user with first_name and last_name" do
+    user = User.new(first_name: "John", last_name: "Doe", email: "john@example.com", password: "password")
+    assert user.save, "Failed to save the user with first_name and last_name"
+  end
```

Comment:
```suggestion
  class UserTest < ActiveSupport::TestCase

  def create_valid_user(first_name, last_name)
    User.new(first_name: first_name, last_name: last_name, email: "john@example.com", password: "password")
  end

  test "should not save user without first_name" do
    user = create_valid_user(nil, "Doe")
    assert_not user.valid?
    assert_includes user.errors[:first_name], "can't be blank"
  end

  test "should not save user without last_name" do
    user = create_valid_user("John", nil)
    assert_not user.valid?
    assert_includes user.errors[:last_name], "can't be blank"
  end

  test "should save user with first_name and last_name" do
    user = create_valid_user("John", "Doe")
    assert user.valid?
    assert user.save, "Failed to save the user with first_name and last_name"
  end

end
```

Reply:
This avoids setting a new user in each test.

It would be a good idea to test if the user doesn't have email as well.

Reply:
Thanks for the suggestion